modules/standard/Regex.chpl

```diff
@@ -586,10 +586,13 @@ proc bytes.this(m:regexMatch) {
 }
 
 
+/* We hold a copy of pattern string/bytes in its serialized form inside
+ * this record.
+ */
 pragma "no doc"
 record chpl_serializeHelper {
   type exprType;
-  var pattern:exprType;
+  var pattern:(new exprType()).chpl__serialize().type;
```

Comment:
IMO this would be better written as

``` chapel
// using a function for this makes the field declaration less complex and allows reuse
private proc getSerializedType(type exprType) type {
  var tmpStr: exprType; // default init of a string variable is more common than `new string()`
  var ser = tmpStr.chpl__serialize();
  return ser.type;
}
record chpl_serializeHelper {
  type exprType;
  var pattern:exprType;
  var pattern: getSerializedType(exprType);
  ...
```

Reply:
much less spicy, but looks good :)